### ISSUE TITLE
WebUI Crashes when logging in with FreeBSD 9 as a host

### ISSUE THREAD

**Issue by thesymbol** [hexparrot/mineos-node#53]
I have tried to install MineOS-node on a FreeBSD box in a jail and everything starts up fine and I can access the WebUI login page via https with the ssh keys generated by the script.
But when I enter the correct details the server silently crashes, i do get redirected but during this time the server silently crashes and i get no output in the console either.

Just to add more info when I enter the wrong credentials I get thrown back to the login page no problem there, the server stays up and dosent crash.

I am running a FreeBSD 9 system, I do not know if it is happening on a FreeBSD 10 system but I would suspect it still crashes.

Is there any extra packages i need to install other than the ones that are required for a debian system?

**Comment by hexparrot** [comment]
FreeBSD isn't yet supported fully (in that the authorization mechanism isn't yet implemented).   I've started working on this recently (https://github.com/hexparrot/mineos-node/tree/freebsd_auth).

FreeBSD support hasn't been highly prioritized yet because it seems that cross-platform inter-operability concerns should follow "getting it working and working right", but thus far, there has been no feedback from the community on how this works on the current target platform, Debian and Debian-derivatives.

**Comment by thesymbol** [comment]
That is good to hear,
I have just tested your freebsd_auth branch and i dont know how far you have gotten with it but when i try and login i get in the console:
```
root@MineOSNode:/usr/games/minecraft # node webui.js
MineOS webui listening on *:8443
undefined
```
I narrowed it down to that it tries to open a shadow file but since freebsd dosent use shadow i do not know how to get any further.
The browser tries to redirect me but is in an infinite loop aswell.

**Comment by hexparrot** [comment]
I think #76 gets FreeBSD operational.  Let me know if you have any issues or any particular functionality doesn't work as expected.

**Comment by thesymbol** [comment]
Looking good so far i only found a few issue, but havent got the time to diagnose if its from my side or if its an issue with mineos, but throwing it out here anyway:
* Creating servers do not work, mineos crashes after a few seconds (like 1-4seconds), no folder is being created for the server, might be a permission issue on my side tho, will look into this tomorrow.
* Another thing that might be a problem but isnt really an issue  is that if you input the wrong password the console outputs undefined and the page keeps "reloading" but not actually reloading so you can still enter your credentials and submit.
* If you download multiple profiles you get Downloaded on all profiles you download when the first one is completed even tho the others have not finished yet. (not related to FreeBSD i think)

**Comment by hexparrot** [comment]
Alright, for the issues you've mentioned

1) creating servers do not work -- this is a huge issue I'd want to tackle, but haven't been able to experience/reproduce it.  Ensure the webui is being run with *root* permissions.
2) fixed this issue just now with 753222cac1f759d67a6afaa58a5afa46ca34d85b.  It will report successful/unsuccessful attempts and return you to the login page (without the 'waiting' to load response you're seeing now).
3) definitely just mineos.  I've been looking into this.
